server/Windows/wf_info.c

```diff
@@ -114,40 +113,61 @@ wfInfo* wf_info_init()
 		DWORD dwType;
 		DWORD dwSize;
 		DWORD dwValue;
-
+
 		wfi->mutex = CreateMutex(NULL, FALSE, NULL);
 
-		if (wfi->mutex == NULL) 
+		if (wfi->mutex == NULL)
 		{
 			WLog_ERR(TAG, "CreateMutex error: %d", GetLastError());
+			free(wfi);
+			return NULL;
 		}
 
 		wfi->updateSemaphore = CreateSemaphore(NULL, 0, 32, NULL);
-
+		if (!wfi->updateSemaphore)
+		{
+			WLog_ERR(TAG, "CreateSemaphore error: %d", GetLastError());
+			CloseHandle(wfi->mutex);
+			free(wfi);
+			return NULL;
+		}
+
 		wfi->updateThread = CreateThread(NULL, 0, wf_update_thread, wfi, CREATE_SUSPENDED, NULL);
 
 		if (!wfi->updateThread)
 		{
 			WLog_ERR(TAG, "Failed to create update thread");
+			CloseHandle(wfi->mutex);
+			CloseHandle(wfi->updateSemaphore);
+			free(wfi);
+			return NULL;
+		}
+
+		wfi->peers = (freerdp_peer**) calloc(1, sizeof(freerdp_peer*) * WF_INFO_MAXPEERS);
+		if (!wfi->peers)
```

Comment:
Could be
wfi->peers = (freerdp_peer\*\*) calloc(WF_INFO_MAXPEERS, sizeof(freerdp_peer*));  ?